daemons/pacemakerd/pacemakerd.c

```diff
@@ -1170,7 +1254,19 @@ main(int argc, char **argv)
     mainloop_add_signal(SIGTERM, pcmk_shutdown);
     mainloop_add_signal(SIGINT, pcmk_shutdown);
 
-    init_children_processes();
+    if ((running_with_sbd) && pcmk__get_sbd_sync_resource_startup()) {
+        crm_notice("Waiting for startup-trigger from SBD.");
+        pacemakerd_state = XML_PING_ATTR_PACEMAKERDSTATE_WAITPING;
+        startup_trigger = mainloop_add_trigger(G_PRIORITY_HIGH, init_children_processes, NULL);
+    } else {
+        if (running_with_sbd) {
+            crm_warn("Enabling SBD_SYNC_RESOURCE_STARTUP would (if supported "
+                     "by your SBD version) improve reliability of "
+                     "interworking between SBD & pacemaker.");
```

Comment:
Absolutely an awesome improvement. Thanks for the nice work!

But in here,  without knowing whether the sbd version even supports the feature, isn't a warning a little too scary for users who don't know about the details? It makes them think something is really wrong here. Would a notice be good enough to deliver the information?

Similar for the warning in sbd:

https://github.com/ClusterLabs/sbd/pull/113/files#diff-cf3c5bf928db3d772ba989e1a462b501782f7e0364daf7414570b14312d90936R981-R983

Reply:
Makes sense

Reply:
Hmm ... my perception of a warning is to point to something that might behave in an unintentional way or lead to sub-optimal behavior ...
And I was fearing that people wouldn't have a look at it if it wasn't at least a warning when doing an upgrade from a previous configuration.
Anyway we should encourage setting the default value to 'yes' (the real default not the value it is set to in the template config) downstream taking care that it is in sync between pacemaker & sbd via package dependencies.
So I can reduce that to a notice together with the next PRs that are touching the topic.

Reply:
Probably we could've bumped the main version of libcrmcommon to make sure sbd built against it would only be able to run with a pacemaker that supports the feature, so that default value of the parameter at build time could've been "auto".

Anyway indeed for the time being, downstream needs to take good care of that.